Add choice statement printer for YANG data

tsyesika opened the pull request:
This adds a printer for schema's which contain a choice field. It's a small nit which was left out of #839 (original issue #835).

Commit:
Add choice statement printer for YANG data

This adds the missing printer to be able to print data which has a
choice field in the schema. This works by taking the leaves in the
choice field case statements and bringing them forward to the same
level as the choice statement, replacing it.

dpino commented on src/lib/yang/data.lua:
```diff
@@ -715,14 +716,28 @@ function data_printer_from_grammar(production)
+         local translator = translators[production.type]
+         if translator ~= nil then
+            local statements = translator(keyword, production)
+            for k,v in pairs(statements) do productions[k] = v end
```
Apparently, it's not OK to add new keys to a table while iterating over it http://lua-users.org/lists/lua-l/2005-09/msg00512.html There was a bug recently related with that https://github.com/Igalia/snabb/pull/831 After reworking the code and not directly adding the new elements to the iterated table, the issue got solved.

Maybe this code is working OK as it is now, but since it was tough to figuring out why #831 was failing, I'd suggest to rework the code in some way to avoid modifying the table while looping it. Removing keys from a table while iterating it seems to be OK though.

tsyesika replied:
Thanks for the heads up. I shall re-work this

Commit:
Fix potential problem modifying a table during iteration

This fixes a potential problem flagged up during review where one
should not add to a table whilst iterating over it. The link to
the mailing list discussion about this is:

http://lua-users.org/lists/lua-l/2005-09/msg00512.html
